### ISSUE TITLE
Compiling issue

### ISSUE THREAD

**Issue by TQMatvey** [mohammad92/android_kernel_samsung_exynos2100#2]
![Screenshot from 2022-02-20 09-36-06](https://user-images.githubusercontent.com/77576395/154826201-23956308-6863-4b84-bc20-2035c8665b35.png)
`drivers/media/platform/exynos/camera/vendor/mcd/is-sec-define.c:1321:14: error: variable length arrays are a C99 feature [-Werror,-Wvla-extension]
        u8 addr_buf[addr_size];
                    ^~~~~~~~~
drivers/media/platform/exynos/camera/vendor/mcd/is-sec-define.c:1369:15: error: variable length arrays are a C99 feature [-Werror,-Wvla-extension]
        u8 write_buf[write_buf_size];
                     ^~~~~~~~~~~~~~
2 errors generated.
make[5]: *** [scripts/Makefile.build:285: drivers/media/platform/exynos/camera/vendor/mcd/is-sec-define.o] Error 1
make[5]: *** Waiting for unfinished jobs....
drivers/media/platform/exynos/camera/vendor/mcd/is-sysfs.c:4966:44: error: this old-style function definition is not preceded by a prototype [-Werror,-Wstrict-prototypes]
int svc_cheating_prevent_device_file_create()
                                           ^
1 error generated.
make[5]: *** [scripts/Makefile.build:285: drivers/media/platform/exynos/camera/vendor/mcd/is-sysfs.o] Error 1
make[4]: *** [scripts/Makefile.build:554: drivers/media/platform/exynos/camera] Error 2
make[3]: *** [scripts/Makefile.build:554: drivers/media/platform/exynos] Error 2
make[2]: *** [scripts/Makefile.build:554: drivers/media/platform] Error 2
make[1]: *** [scripts/Makefile.build:554: drivers/media] Error 2
make: *** [Makefile:1835: drivers] Error 2
make: Leaving directory '/home/matvey/kernel/afaneh/android_kernel_samsung_exynos2100'
Kernel STUCK in BUILD!, Press [Enter] key to return to Main menu...
`

i always had same issue on debian 11, archj linux (now), ubuntu....
Trying to compile for p3s but other devices have same issue

**Comment by emaxoda** [comment]
How did you solve this problem?

**Comment by utkustnr** [comment]
> How did you solve this problem?

I'm few months late to the party but since this is the first place that shows up when I searched my issue I figured I should give an answer to this. My problem wasn't in this tree but I had the same function erroring out as I'm dealing with samsung as well. Changing `int svc_cheating_prevent_device_file_create()` to `int svc_cheating_prevent_device_file_create(void)` fixed it. Best of luck.
